bweapons_api/api.lua

```diff
@@ -406,28 +406,61 @@ function bweapons.register_weapon(def)
             end
 
             if def.has_durability then
-                local meta = minetest.deserialize(itemstack:get_metadata())
-                if
-                def.requires_technic and not meta or
-                def.requires_technic and meta.charge < technic_charge_per_use or
-                not def.requires_technic and def.custom_charge and (65535 - itemstack:get_wear()) < (65535 / uses)
-                then
+                local reload = false
+
+                if def.requires_technic then
+                    -- Technic Plus Beta
+                    if technic.use_RE_charge ~= nil then
+                        if not technic.use_RE_charge(itemstack, technic_charge_per_use) then
+                            reload = true
+                        end
+                    else
+                        -- Technic Plus
+                        local meta = itemstack:get_meta()
+                        local old_metadata = minetest.deserialize(meta:get_string(""))
+                        local charge = nil
+                        if old_metadata then
+                            charge = old_metadata.charge
+                        end
+                        -- Technic
+                        if not charge then
+                            charge = meta:get_int("technic:charge")
+                        end
+
+                        if charge < technic_charge_per_use then
+                            reload = true
+                        else
+                            if not technic.creative_mode then
+                                charge = charge - technic_charge_per_use
+                                technic.set_RE_wear(itemstack, charge, technic_charge)
+                                if old_metadata and old_metadata.charge then
+                                    old_metadata.charge = charge
+                                    meta:set_string("", minetest.serialize(old_metadata))
+                                else
+                                    meta:set_int("technic:charge", charge)
+                                end
+                            end
+                        end
```

Comment:
I think this would be enough for both migration and usage with original Technic, Technic Plus stable and Technic Plus Beta if we get https://github.com/mt-mods/technic/pull/391 through:

```suggestion
                        -- Technic, old metadata migration
                        local meta = itemstack:get_meta()
                        local old_metadata = meta:get("")
                        if old_metadata and old_metadata.charge then
                            technic.set_charge(itemstack, old_metadata.charge)
                            meta:set_string("", "")
                        end

                        -- Technic, check and use charge
                        local charge = technic.get_charge(itemstack)
                        if charge < technic_charge_per_use then
                            reload = true
                        elseif not technic.creative_mode then
                            technic.set_charge(itemstack, charge - technic_charge_per_use)
                        end
```
That's already +15 -26 LoC

Considering simplicity of changes and compatibility gains on Technic Plus side for me this seems significant improvement.

Reply:
Also any Technic fork that implement technic.get_charge and technic.set_charge

Though if any of the forks with these are still using old meta then migration would do unnecessary metadata reset each time.

Also I'm not sure if full old metadata reset is safe with these items, if something else round here uses old metadata then that something else would be lost too. However I'd argue full cleanup of old metadata would be very good in case it isn't needed.